SI-6388 Removes deprecated functionality for 2.11

soc opened the pull request:
Not in this commit:
- Deprecated methods of List (does Paul still want to keep them?)
- @serializable annotation (removal needs multiple steps)

soc committed:
SI-6388 Remove first parts of deprecated @serializable annotation

soc commented on src/compiler/scala/tools/nsc/Global.scala:
```diff
@@ -1733,20 +1733,12 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
-  def forInteractive   = onlyPresentation
-  def forScaladoc      = onlyPresentation
+  def forInteractive   = false
+  def forScaladoc      = false
```
Is this the right solution?

gkossakowski replied:
@VladUreche: can you comment on this fragment?

VladUreche replied:
This change LGTM, it's overridden in DocFactory anyway.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/496/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1204/

soc committed:
SI-6388 Remove deprecated items in scala/collection

scala-jenkins commented:
jenkins job pr-rangepos: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/496/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1208/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1204/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1208/

soc commented:
I just tested it without the change in Entity.scala:L153 and it still fails ...

soc commented:
The cause is in the first commit, but it is not `inheritedFrom` ... weird.

paulp commented:
Some of these need to be kept longer.  At least: src/compiler/scala/tools/nsc/Interpreter.scala,InterpreterLoop.scala.  Have only started looking at it.

paulp commented:
Okay, I think that's it.  The List methods can go.  The rest LGTM but a second pair of eyes would be good.

gkossakowski commented:
I'd prefer if we had more fine-grained commits. The bare minimum is to split changes made to compiler and library. The best would be that you go one package at the time. The reason is that I want to be able to easily pinpoint the exact change that broke some tool and revert it if needed without reverting everything else.

For this kind of changes I totally don't mind having 5 or 10 focused commits. I actually prefer that.

soc commented:
@gkossakowski You're absolutely right. I'm currently splitting up the commits into smaller parts and test them individually.

As a first benefit of having smaller commits, I was able to reduce the amount of suspects involved in the ScalaDoc test failure. It really looks like as if either the removal of `Application` or the items in `Predef` are causing the error.

I still have no idea how this happens, though ...

soc committed:
SI-6388 Remove deprecated items in scala/math

soc committed:
SI-6388 Remove deprecated items in the compiler jar

As requestied by Paul, src/compiler/scala/tools/nsc/Interpreter.scala
and src/compiler/scala/tools/nsc/InterpreterLoop.scala are kept.

soc committed:
SI-6388 Remove some remaining, minor deprecations
